### ISSUE TITLE
Crippled Vx Connectbot?

### ISSUE THREAD

**Issue by compata** [vx/connectbot#74]
Version 1.7.1-29 on SGS3 from Play Store.

Menu offers SSH, telnet, and local. No SCP.

I can't find any setting to permit screen rotation while connected. Without rotation, several keyboard keys are not visible, including CTRL. Not helpful.

**Comment by mmatuska** [comment]
SCP is only available from an existing SSH session. Browsing the remote host is not implemented. I don't really know about the screen rotation problem, very much depends on the Android version and phone model. The highest version I have developed this on is Android 4.0.4 and from my side development is frozen atm.

**Comment by compata** [comment]
Thank you. (1) So to copy a file TO the phone, I have to SSH to the file's location, then initiate an SCP copy from that machine back to the phone? (2) Only Vx Connectbot refuses to rotate. All other applications, and the initial connection dialog in Vx rotate fine. I recall something during installation about not rotating the screen but, of course, I can't easily get back to that.

**Comment by Enekk** [comment]
Compata, I don't know about your rotation issues, but I do know the answer to your scp question.  Once you have connected to a remote host, you should be able to bring up the drop down menu (you might have to single tap on the terminal window to get the menu bar to pop up).  From there, you can select "Download File".  I've had mixed success with the paths that it wants, so I generally give the full path to the file I'm trying to download (and I avoid using shortcuts like '~' for home).  Your results may vary.  I have attached a screen shot of the menu with the appropriate option selected (mine is greyed out because this was just a localhost connection, it usually won't be).

![2013-07-11 06 58 06](https://f.cloud.github.com/assets/1928133/781179/f61389fc-ea19-11e2-8ca8-f242ea299d54.png)

Edit: Wow, apparently you can't decide on the size of an image when adding it to a post!  Sorry for that.

**Comment by compata** [comment]
Thanks,  Enekk. But this just returns to my rotation problem. I can see the "fullscreen" button but not the drop-down symbol to the right. It's just cut off on my screen. I could probably see it if I could rotate the screen.

**Comment by Enekk** [comment]
Have you got to the settings in the main application and gone to "Rotation Mode"?  There is a set of options there that might help you.  I'm sure you've seen it, but if not maybe playing with those will help?

**Comment by compata** [comment]
I think that is exactly what I need. But as I wrote initially, "I can't find any setting to permit screen rotation while connected." Could you please be so kind as to tell me the precise steps needed to reach this option?

Of course, if you refer to the phone's main settings screen, then I've already looked at that. Rotation is enabled and, as I said, works properly for all other applications. There must be something about this one that is different.

**Comment by Enekk** [comment]
i was talking about the phone's main settings screen.  Perhaps you could try some of the other options instead of allow rotation?  What happens if you select the force landscape option? 

Also, from what research I did, it seems like with the Galaxy S3 and spotty rotation are not uncommon.  Solutions are all over the place and I'm certainly not qualified to tell you which ones to try, so I'll leave searching for that sort of thing up to you.

One other thing you could do is completely remove ConnectBot (make sure any of its data folders go away too - no I don't know where they all live) and then reinstall it.  See if that at least fixes the rotation issue.  If you have TitaniumBackup or similar, you should make a backup of the app before you do this.

**Comment by compata** [comment]
Wonderful! I've installed the "screen rotation control" app and now I can see this one in landscape mode. I wouldn't have guessed such an app was necessary given that the phone already has rotation settings. So thank you for the idea.

But, interestingly, even in landscape mode, the "fullscreen" button is flush right. I still can't get to the drop-down menu. At least I can now see the entire keyboard.

**Comment by Enekk** [comment]
Yeah, that is odd.  It almost sounds like Connectbot is seeing your resolution as larger than it actually is.  Not sure why that would be.  Glad you found a solution though.

**Comment by mzilikazi** [comment]
I have also encountered this issue on a Transformer Prime TF201 running Pacman ROM 4.3Build-1

Go to: Settings > PAC Man settings > General UI > Show Menu UI overflow

This was not enabled which meant that the 3 dot menu on the top right of the VX Connectbot app was not visible.
Note that it was also required to be actively within a session and long press the screen to bring up the 'Full screen mode' / Change font size menu -> Enable full screen mode. Then a short press would show the FULLSCREEN : menu in the top right corner.
